Gems/GradientSignal/Code/Source/Editor/EditorImageGradientComponent.cpp

```diff
@@ -57,14 +73,144 @@ namespace GradientSignal
+        AzToolsFramework::PaintBrushComponentRequestBus::Handler::BusConnect(GetEntityId());
+
         m_componentModeDelegate.ConnectWithSingleComponentMode<EditorImageGradientComponent, EditorImageGradientComponentMode>(
             AZ::EntityComponentIdPair(GetEntityId(), GetId()), nullptr);
     }
 
+    void EditorImageGradientComponent::OnAssetReady(AZ::Data::Asset<AZ::Data::AssetData> asset)
+    {
+        if (!m_configuration.m_overrideAsset.Get())
+        {
+            AZ::Data::AssetId newAssetId = AZ::Data::AssetId(AZ::Uuid::CreateRandom());
+            AZ::Data::Asset<ImageAsset> imageAsset = asset;
+            m_configuration.m_overrideAsset = AZ::Data::AssetManager::Instance().CreateAsset(
+                newAssetId, asset.GetType(), m_configuration.m_overrideAsset.GetAutoLoadBehavior());
+
+            m_component.CopyImageAsset(imageAsset, m_configuration.m_overrideAsset);
+        }
+    }
+
     void EditorImageGradientComponent::Deactivate()
     {
         m_componentModeDelegate.Disconnect();
+
+        AZ::Data::AssetBus::Handler::BusDisconnect();
+        AzToolsFramework::PaintBrushComponentRequestBus::Handler::BusDisconnect();
+
         m_paintBrush.Deactivate();
         BaseClassType::Deactivate();
     }
+
+    void EditorImageGradientComponent::OnCompositionChanged()
+    {
+        BaseClassType::OnCompositionChanged();
+
+        if (!AZ::Data::AssetBus::Handler::BusIsConnectedId(m_configuration.m_imageAsset.GetId()))
+        {
+            m_configuration.m_useOverride = false;
+            m_configuration.m_overrideAsset = { AZ::Data::AssetLoadBehavior::QueueLoad };
+            m_path = "";
+
+            m_component.ClearOverrideConfiguration();
+
+            AZ::Data::AssetBus::Handler::BusDisconnect();
+            AZ::Data::AssetBus::Handler::BusConnect(m_configuration.m_imageAsset.GetId());
+        }
+    }
+
+    void EditorImageGradientComponent::SavePaintLayer()
+    {
+        char projectPath[AZ_MAX_PATH_LEN];
+        AZ::IO::FileIOBase::GetInstance()->ResolvePath("@devassets@", projectPath, AZ_MAX_PATH_LEN);
+
+        AZStd::string relativePath = m_path;
+        AZStd::string fullPath;
+
+        if (!relativePath.empty())
+        {
+            AzFramework::StringFunc::Path::Join(projectPath, relativePath.c_str(), fullPath, true, true);
+
+            if (!AZ::IO::FileIOBase::GetInstance()->Exists(fullPath.c_str()))
+            {
+                relativePath.clear();
+            }
+        }
+
+        if (relativePath.empty())
+        {
+            AZ::Uuid uuid = AZ::Uuid::CreateRandom();
+            AZStd::string uuidString;
+            uuid.ToString(uuidString);
+
+            relativePath = "ImageGradientOverride/" + uuidString + "_gsi.dds";
+
+            auto invalidCharacters = [](char letter)
+            {
+                return letter == ':' || letter == '"' || letter == '\'' || letter == '{' || letter == '}' || letter == '<' || letter == '>';
+            };
```

Comment:
This can be simplified to use the C++23 [contains](https://en.cppreference.com/w/cpp/string/basic_string_view/contains) function which has been implemented in AZStd
```suggestion
            auto invalidCharacters = [](char letter)
            {
                return AZStd::string_view(R"(:"'{}<>)").contains(letter);
            };
```